pkg/web/routes.go

```diff
@@ -686,11 +687,26 @@ func (wa *WebAPI) buildPipelines(
 		// Set status to pending while we process modules
 		setCommitStatusByAction(p, s.InstanceId, git.StatusPending, builder.Action)
 
-		ignoreFileRegExps := getIgnoreFileRegExps(p, d, l)
+		var ignoreFileRegExps []*regexp2.Regexp
+		var ignoreFilePatterns []string
+
+		if s.DinghyIgnoreRegexp2Enabled {
```

Comment:
What we could do as well:
1. Create an type IgnoreFile interface, that exposes a method like shouldIgnore
2. Create two implementations of that interface - one that executes the filtering as before, one that does it the new way,
3. Based on the value of s.DinghyIngnoreRegexp2Enabled we can select implantation to instantiate, and then use it later with no additional checks for the value of the setting.

The benefit I see in such approach is that we can easily unit test IgnoreFile implementations with minimal test dependencies.

Of course, changes are great as they are now. I just wanted to share an idea that came to my mind as I was checking the PR. :)

Well done, Alexey!